src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,343 @@
+        return deadlineAdded;
+    }
+
+    // marks tasks as done
+    public static void markTaskAsDone(String userInput) {
+        userInput = userInput.replaceAll("[^0-9]", "");
+        try {
+            int taskNo = Integer.parseInt(userInput);
+            // handle error where task no is out of range
+            if (taskNo > arrayOfTasks.size()) {
+                System.out.println("You only have " + arrayOfTasks.size() + " task(s)!");
+            } else {
+                arrayOfTasks.get(taskNo - 1).markAsDone();
+                System.out.println("Nice! I have marked this task as done:" + arrayOfTasks.get(taskNo - 1));
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid Input! Input format should have an integer e.g. done 2");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid Input! Integer cannot be 0!");
+        } finally {
+            System.out.println(HORIZONTAL_LINE);
+        }
+    }
+
+    // Delete tasks
+    private static void deleteTask(String userInput) {
+        userInput = userInput.replaceAll("[^0-9]", "");
+        try {
+            int taskNo = Integer.parseInt(userInput);
+            // handle error where task no is out of range
+            if (taskNo > arrayOfTasks.size()) {
+                System.out.println("You only have " + arrayOfTasks.size() + " task(s)!");
+            } else {
+                System.out.println("Noted. I've removed this task: " + System.lineSeparator() + arrayOfTasks.get(taskNo - 1));
+                arrayOfTasks.remove(taskNo - 1);
+                System.out.println("Now you have " + arrayOfTasks.size() + " tasks in the list.");
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid Input! Input format should have an integer e.g. done 2");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid Input! Integer cannot be 0!");
+        } finally {
+            System.out.println(HORIZONTAL_LINE);
+        }
+    }
+
+
+    // prints items on current arrayOfTasks
+    public static void viewTasks(ArrayList<Task> arrayOfTasks) {
+        // handle error case where no tasks in array
+        if (arrayOfTasks.size() == 0) {
+            System.out.println("There are no tasks in your list.");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 1; i <= arrayOfTasks.size(); i++) {
+                System.out.println(i + ". " + arrayOfTasks.get(i - 1));
+            }
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    // ends program when reader says 'bye'
+    public static int sayBye(String byeInput) {
+        if (byeInput.toLowerCase().contains(KEYWORD_BYE)) {
+            System.out.println("Bye. Hope to see you again soon!");
+            System.out.println(HORIZONTAL_LINE);
+            return 1;
```

Comment:
Maybe could change to return true/false, as using 1/0 is less clear from a readability standpoint